### DISCUSSION TITLE
Is there a way to make an Android Widget?

### DISCUSSION THREAD

**Discussion by eamonburns** [fyne-io/fyne#4759] [Q&A] [answered]
(I am not talking about `fyne.Widget`)

Is there a way to create widgets that a user can add to their phone's home screen?

**Comment by andydotxyz** [comment, marked as answer]
We don't have support for OS/dashboard widgets.
Would be cool though.

**Reply by eamonburns** [reply]
Ok, thanks for the answer. I wonder if there would be a way to make an extension that allows you to create them? (like the ones listed on [fyne-io/fyne-x](https://github.com/fyne-io/fyne-x))

There is [official documentation](https://developer.android.com/develop/ui/views/appwidgets) about widgets, so maybe there could be a way to "register" a Fyne window as a widget? (The exact definition of app widgets is this: "App widgets are miniature app views that you can  embed in other apps—such as the home screen—and receive periodic updates")

**Reply by eamonburns** [reply]
I know next to nothing about Android development though, so I wouldn't be the one to figure it out unfortunately

**Reply by andydotxyz** [reply]
If you read further into the Android documentation that you linked you will see that it relies on a "View XML" so it's not simply a "here is my window, embed it" operation.
The following question answered something that I expect to be closer to our reality at this time: https://stackoverflow.com/questions/12682822/using-open-gl-in-an-android-widget

**Reply by eamonburns** [reply]
@andydotxyz are there any examples of Fyne interacting with the Android API using XML like described in the documentation? I assume the documentation is for Java

**Reply by andydotxyz** [reply]
No, we don't have any APIs for the Android XML render information as we provide a different render stack (based on OpenGL, hence my link above).

**Reply by eamonburns** [reply]
Ah, well that is unfortunate
